test/FullRange.t.sol

```diff
@@ -0,0 +1,767 @@
+        token2.mint(address(this), 2 ** 128);
+
+        manager = new PoolManager(500000);
+
+        FullRangeImplementation impl = new FullRangeImplementation(manager, fullRange);
+        vm.etch(address(fullRange), address(impl).code);
+
+        key = PoolKey(Currency.wrap(address(token0)), Currency.wrap(address(token1)), 3000, TICK_SPACING, fullRange);
+        id = key.toId();
+
+        key2 = PoolKey(Currency.wrap(address(token1)), Currency.wrap(address(token2)), 3000, TICK_SPACING, fullRange);
+        id2 = key.toId();
+
+        withLiqKey =
+            PoolKey(Currency.wrap(address(token0)), Currency.wrap(address(token2)), 3000, TICK_SPACING, fullRange);
+        withLiqId = withLiqKey.toId();
+
+        modifyPositionRouter = new PoolModifyPositionTest(manager);
+        swapRouter = new PoolSwapTest(manager);
+
+        token0.approve(address(fullRange), type(uint256).max);
+        token1.approve(address(fullRange), type(uint256).max);
+        token2.approve(address(fullRange), type(uint256).max);
+        token0.approve(address(swapRouter), type(uint256).max);
+        token1.approve(address(swapRouter), type(uint256).max);
+        token2.approve(address(swapRouter), type(uint256).max);
+
+        manager.initialize(withLiqKey, SQRT_RATIO_1_1);
+        fullRange.addLiquidity(
+            FullRange.AddLiquidityParams(
+                address(token0),
+                address(token2),
+                3000,
+                100 ether,
+                100 ether,
+                99 ether,
+                99 ether,
+                address(this),
+                MAX_DEADLINE
+            )
+        );
+    }
+
+    function testFullRange_beforeInitialize_AllowsPoolCreation() public {
+        PoolKey memory testKey = key;
+
+        vm.expectEmit(true, true, true, true);
+        emit Initialize(id, testKey.currency0, testKey.currency1, testKey.fee, testKey.tickSpacing, testKey.hooks);
+
+        snapStart("FullRangeInitialize");
+        manager.initialize(testKey, SQRT_RATIO_1_1);
+        snapEnd();
+
+        (, address liquidityToken) = fullRange.poolInfo(id);
+
+        assertFalse(liquidityToken == address(0));
+    }
+
+    function testFullRange_beforeInitialize_RevertsIfWrongSpacing() public {
+        PoolKey memory wrongKey =
+            PoolKey(Currency.wrap(address(token0)), Currency.wrap(address(token1)), 0, TICK_SPACING + 1, fullRange);
+
+        vm.expectRevert(FullRange.TickSpacingNotDefault.selector);
+        manager.initialize(wrongKey, SQRT_RATIO_1_1);
+    }
+
+    function testFullRange_addLiquidity_InitialAddSucceeds() public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        uint256 prevBalance0 = MockERC20(token0).balanceOf(address(this));
+        uint256 prevBalance1 = MockERC20(token1).balanceOf(address(this));
+
+        address token0Addr = address(token0);
+        address token1Addr = address(token1);
+
+        snapStart("FullRangeAddLiquidity");
```

Comment:
can we name this `FullRangeAddInitialLiquidity` and have another gas test `FullRangeAddLiquidity` that adds to the keyWithLiq since that's a more accurate gas snapshot for average LPs?